x-pack/plugins/security/server/routes/authentication/common.ts

```diff
@@ -123,66 +132,70 @@ export function defineCommonRoutes({
+                schema.literal(TokenAuthenticationProvider.type),
+              ]),
+              basicParamsSchema,
+              schema.never()
+            ),
+          }),
+        },
+        options: { authRequired: false },
       },
-      options: { authRequired: false },
-    },
-    createLicensedRouteHandler(async (context, request, response) => {
-      const { providerType, providerName, currentURL, params } = request.body;
-      logger.info(`Logging in with provider "${providerName}" (${providerType})`);
-
-      const redirectURL = parseNext(currentURL, basePath.serverBasePath);
-      const authenticationResult = await getAuthenticationService().login(request, {
-        provider: { name: providerName },
-        redirectURL,
-        value: getLoginAttemptForProviderType(providerType, redirectURL, params),
-      });
-
-      if (authenticationResult.redirected() || authenticationResult.succeeded()) {
-        return response.ok({
-          body: { location: authenticationResult.redirectURL || redirectURL },
-          headers: authenticationResult.authResponseHeaders,
+      createLicensedRouteHandler(async (context, request, response) => {
+        const { providerType, providerName, currentURL, params } = request.body;
+        logger.info(`Logging in with provider "${providerName}" (${providerType})`);
+
+        const redirectURL = parseNext(currentURL, basePath.serverBasePath);
+        const authenticationResult = await getAuthenticationService().login(request, {
+          provider: { name: providerName },
+          redirectURL,
+          value: getLoginAttemptForProviderType(providerType, redirectURL, params),
         });
-      }
 
-      return response.unauthorized({
-        body: authenticationResult.error,
-        headers: authenticationResult.authResponseHeaders,
-      });
-    })
-  );
-
-  router.post(
-    { path: '/internal/security/access_agreement/acknowledge', validate: false },
-    createLicensedRouteHandler(async (context, request, response) => {
-      // If license doesn't allow access agreement we shouldn't handle request.
-      if (!license.getFeatures().allowAccessAgreement) {
-        logger.warn(`Attempted to acknowledge access agreement when license doesn't allow it.`);
-        return response.forbidden({
-          body: { message: `Current license doesn't support access agreement.` },
+        if (authenticationResult.redirected() || authenticationResult.succeeded()) {
+          return response.ok({
+            body: { location: authenticationResult.redirectURL || redirectURL },
+            headers: authenticationResult.authResponseHeaders,
+          });
+        }
+
+        return response.unauthorized({
+          body: authenticationResult.error,
+          headers: authenticationResult.authResponseHeaders,
         });
-      }
+      })
+    );
+
+    // In the serverless environment, there is no need for access agreement acknowledgment
```

Comment:
Maybe something like this (to make it clear why there is no need this in the specific API)?
```suggestion
    // In the serverless offering, the access agreement functionality isn't available.
```